ssh/tailssh/incubator.go

```diff
@@ -222,13 +222,23 @@ func beIncubator(args []string) error {
 		}
 	}
 
+	homeDir := "/"
+	if userMeta, err := userLookupGetent(ia.localUser); err == nil {
+		if _, err := os.Stat(homeDir); err == nil || os.IsPermission(err) {
+			homeDir = userMeta.HomeDir
+		}
+
+	}
+
 	euid := os.Geteuid()
 	runningAsRoot := euid == 0
 	if runningAsRoot && ia.loginCmdPath != "" {
 		// Check if we can exec into the login command instead of trying to
 		// incubate ourselves.
 		if la := ia.loginArgs(); la != nil {
-			return unix.Exec(ia.loginCmdPath, la, os.Environ())
+			// PWD is used by unix.Exec to set the directory
```

Comment:
That's not true. Not in Go (see the code in x/sys/unix and syscall) nor documented in the exec manpages.

Reply:
I guess you mean by the thing that was execed (the login cmd).

Reply:
Yes, was referring to login cmd. Based on interactive and non-interactive commands it worked. I can test SSHD as well.